src/main/java/com/amuzil/omegasource/magus/skill/conditionals/InputDataBuilder.java

```diff
@@ -0,0 +1,36 @@
+package com.amuzil.omegasource.magus.skill.conditionals;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Designed as a factory class for taking InputDat and turning it into a linked list
+ * for InputModule use.
+ */
+public class InputDataBuilder {
+
+    public static LinkedList<InputData> toInputs(InputData... data) {
+        LinkedList<InputData> out = new LinkedList<>();
+        Collections.addAll(out, data);
+        return out;
+    }
+
+    public static InputPermutation createPermutation(InputData... inputs) {
+        LinkedList<InputData> data = new LinkedList<>(List.of(inputs));
+        return new InputPermutation(data);
```

Comment:
What's the purpose of having an intermediate variable here?

Reply:
Easy fix, I'll get on that